service/notesservice.php

```diff
@@ -196,6 +181,31 @@ public function delete ($id, $userId) {
         $file->delete();
     }
 
+    private function getSafeTitleFromContent($content) {
+        // prepare content: remove markdown characters and empty spaces
+        $content = preg_replace("/^\s*[*+-]\s+/m", "", $content); // list item
+        $content = preg_replace("/^#+\s+(.*?)\s*#*$/m", "$1", $content); // headline
+        $content = preg_replace("/^(?:=*|-*)$/m", "", $content); // separate line for headline
```

Comment:
I think, you're right. It's a long time ago, when I wrote those regular expressions. Therefore, I actually don't remember why I chose that variant. I will test your suggestion in detail and change it if it works.